mergify_engine/web.py

```diff
@@ -151,6 +160,67 @@ def subscription_cache(installation_id):  # pragma: no cover
     return "Cache cleaned", 200
 
 
+SimulatorSchema = voluptuous.Schema({
+    voluptuous.Required('url'): voluptuous.Url(),
+    voluptuous.Required('mergify.yml'): str,
+})
+
+
+def abort(message, code=400):
+    flask.abort(flask.make_response(
+        flask.jsonify(message=message), code
+    ))
+
+
+@app.route("/simulator", methods=["POST"])
+def simulator():
+    authentification()
+
+    try:
+        data = SimulatorSchema(flask.request.get_json(force=True))
+    except voluptuous.MultipleInvalid as e:
+        abort("Invalid input: %s" % str(e))
+
+    try:
+        _, owner, repo, _, pull_number = urlsplit(data["url"]).path.split("/")
+        pull_number = int(pull_number)
+    except ValueError:
+        abort("Invalid pull request url")
+
+    integration = github.GithubIntegration(config.INTEGRATION_ID,
+                                           config.PRIVATE_KEY)
+    installation_id = utils.get_installation_id(integration, owner)
+    if not installation_id:  # pragma: no cover
+        abort("Mergify not installed")
+
+    token = integration.get_access_token(installation_id).token
+    pull = mergify_pull.MergifyPull.from_number(
+        installation_id, token, owner, repo, pull_number
+    )
+
+    try:
+        mergify_config = rules.validate_user_config(data["mergify.yml"]) or {}
+    except rules.InvalidRules as e:  # pragma: no cover
+        message = str(e)
+        if hasattr(e, 'problem_mark'):
+            message += " (position %s:%s)" % (e.problem_mark.line + 1,
```

Comment:
Rather than `abort()`ing with that message, I'm pretty sure that exposing those position as a proper exception field that would be serialized (as suggested earlier) would ease error reporting on the UI side.

Reply:
I created a new voluptuous object to handle this cleanly.